### PR TITLE
M1758

### PR DESCRIPTION
**Pull request by aspcanada**

### CODE REVIEW

File: src/api/reports/attributes_report.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        "Vulnerability",
+        "Climate Score",
+    ] + region_names
+
+    data = [
+        COLUMN_NAMES,
+        *[
+            [
+                fish_species.genus.family.name,
+                fish_species.genus.name,
```
Comment (saanobhaai) [inline src/api/reports/attributes_report.py:163]: I don't think this is right... I think `COLUMN_NAMES` is correct, but `genus.name` should be removed here, and genus + species.name should be concatenated here -- see `get_display_name` in `FishSpeciesSerializer`

File: src/api/reports/attributes_report.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                fish_species.max_length_type,
+                fish_species.group_size and fish_species.group_size.name,
+                fish_species.trophic_level,
+                fish_species.vulnerability,
+                fish_species.climate_score,
+                *create_regions_row(
+                    regions, [str(r.id) for r in fish_species.regions.all()]
+                ),
+            ]
+            for fish_species in FishSpecies.objects.select_related(
```
Comment (saanobhaai) [inline src/api/reports/attributes_report.py:180]: Compare queryset used in `FishSpeciesViewSet` -- seems like this should match (or even both use one definition?)

**Review by saanobhaai — CHANGES_REQUESTED** [review]

**Comment by saanobhaai** [comment]
[fish and benthic attribute xlsx endpoint](https://trello.com/c/RAhb2AyH/1758-fish-and-benthic-attribute-xlsx-endpoint)

File: iac/stacks/static_site.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                bucket=site_bucket, 
+                origin_access_identity=cloudfront_OAI
+            ),
+            allowed_methods=cf.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
+            viewer_protocol_policy=cf.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+        )
+
+        distribution = cf.Distribution(
+            self,
+            id="Distribution",
```
[the remark this answers is not in the document]
Comment (aspcanada) [reply iac/stacks/static_site.py:93]: Nah, it's how we're doing it for the webapp. The ID gets attached to the cloudformation stack (see attached). 
Screenshot 2023-01-20 at 9 58 32 AM

File: iac/stacks/static_site.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+from iac.settings import ProjectSettings
+
+
+class StaticSiteStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        config: ProjectSettings,
+        # api_zone: r53.HostedZone,
```
[the remark this answers is not in the document]
Comment (aspcanada) [reply iac/stacks/static_site.py]: done

File: iac/stacks/static_site.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    aws_route53 as r53,
+    aws_route53_targets as targets,
+    aws_certificatemanager as acm,
+    aws_cloudfront as cf,
+    aws_cloudfront_origins as cf_origins,
+    aws_iam as iam,
+    aws_s3 as s3,
+    aws_s3_deployment as s3Deploy,
+)
+# from static_site import StaticSitePublicS3, StaticSitePrivateS3
```
[the remark this answers is not in the document]
Comment (aspcanada) [reply iac/stacks/static_site.py]: done

File: src/api/reports/attributes_report.py [hunk 6]
```diff
@@ -0,0 +1,262 @@
+from collections import defaultdict
+
+from openpyxl import Workbook
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/reports/attributes_report.py:3]: We aren't there yet on using the other library, can refactor if/when we switch.

File: src/api/utils/s3.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import boto3
```
[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/utils/s3.py:1]: I think we can refactor file name if/when needed.  The queue is using boto for other things but that is a separate django app and should be kept separate from the API.  I agree that the other S3 calls should use the same code (db backup/restore scripts).  I can create a issue for that

[the remark this answers is not in the document]
Comment (gridcell) [reply src/api/utils/s3.py:1]: https://trello.com/c/8J4uViey/1814-refactor-the-s3-calls-with-dbrestore-and-dbbackup-to-use-s3py-module

File: src/api/reports/attributes_report.py (hunk 2 above)
Comment (gridcell) [reply src/api/reports/attributes_report.py:180]: I matched the quersets

**Pull request merged**